src/main/java/io/rtr/conduit/amqp/impl/AMQPTransport.java

```diff
@@ -177,6 +187,18 @@ protected void stopImpl() throws IOException {
             } catch (AlreadyClosedException ignored) {
             }
         }
+        if (executor != null) {
+            executor.shutdown();
+        }
+    }
+
+    @Override
+    protected boolean isStoppedImpl(int waitForMillis) {
+        try {
+            return executor.awaitTermination(waitForMillis, TimeUnit.MILLISECONDS);
+        } catch (InterruptedException e) {
+            return false; // interrupted while waiting for termination
```

Comment:
Do we need to reinterrupt here?

Reply:
Yeah, that's probably more prudent. I was resisting propagating the checked exception to the caller, but in this case, I agree it's the right thing to do.